doniclient/osc/cli.py

```diff
@@ -115,246 +74,226 @@ def take_action(self, parsed_args):
         )
 
 
-class DeleteHardware(BaseParser):
+class DeleteHardware(SingleParser):
     """Delete specific hardware item in Doni."""
 
-    require_hardware = True
-
     def take_action(self, parsed_args):
         hw_client = self.app.client_manager.inventory
         try:
-            result = hw_client.delete(parsed_args.uuid)
+            hw_client.delete(parsed_args.uuid)
         except HttpError as ex:
             LOG.error(ex.response.text)
             raise ex
 
-        return result.text
-
 
-class SyncHardware(BaseParser):
+class SyncHardware(SingleParser):
     """Sync specific hardware item in Doni."""
 
-    require_hardware = True
-
     def take_action(self, parsed_args):
         hw_client = self.app.client_manager.inventory
         try:
-            result = hw_client.sync(parsed_args.uuid)
+            hw_client.sync(parsed_args.uuid)
         except HttpError as ex:
             LOG.error(ex.response.text)
             raise ex
 
-        return result.text
-
-
-class CreateHardware(BaseParser, HardwareSerializer):
-    """Create a Hardware Object in Doni."""
 
+class CreateOrUpdateParser(BaseParser):
     def get_parser(self, prog_name):
         parser = super().get_parser(prog_name)
         parser.add_argument(
             "--name",
-            metavar="<name>",
             help=(
                 "Name of the hardware object. Best practice is to use a "
                 "universally unique identifier, such has serial number or chassis ID. "
                 "This will aid in disambiguating systems."
             ),
-            required=True,
         )
         parser.add_argument(
             "--hardware_type",
-            metavar="<hardware_type>",
-            help=("hardware_type of item"),
             default="baremetal",
+            help=("hardware_type of item"),
         )
-        parser.add_argument("--mgmt_addr", metavar="<mgmt_addr>", required=True)
-        parser.add_argument("--ipmi_username", metavar="<ipmi_username>")
-        parser.add_argument("--ipmi_password", metavar="<ipmi_password>")
+
         parser.add_argument(
-            "--ipmi_terminal_port", metavar="<ipmi_terminal_port>", type=int
+            "--properties",
```

Comment:
What happens when this is provided in conjunction w/ other flags, does it get merged in? What takes precedence?